### PR TITLE
Updates to sql_repo branch

### PR DESCRIPTION
**Pull request by david4096**
Added documentation to configuration for optional arguments and fixed a few typos.

Changing references of `add-ontology` to `add-ontologymap` and reinstating some end to end tests. Removed calls to `destroy`.

The tests for the repo need to be for a single reference set, so the test BAM files have been switched to those from thousand genomes and NCBI37.

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -40,8 +41,8 @@ the first command you run.
 verify
 +++++++
 
-Performs some consistency checks on the given data repository to ensure it is
-well-formed.
+Checks that the data pointed to in the repository works and
+there aren't any broken URLs or missing files.
```
Comment (macieksmuga) [inline docs/configuration.rst]: plural data? Alternately:
```
Checks that the data files pointed to in the repository exist and are readable in the locations specified by file paths or URLs.
```

**Comment by macieksmuga** [comment]
+1 These look good to me (modulo minor wording edit above). All tests pass cleanly.

**Comment by jeromekelleher** [comment]
-1

The reasoning for Ontology has been discussed: https://github.com/ga4gh/server/pull/1117 and https://github.com/ga4gh/server/issues/1147

**Comment by macieksmuga** [comment]
I'm officially confused.

**Comment by david4096** [comment]
@jeromekelleher updated, thanks!

File: tests/unit/test_cli.py [hunk 2]
```diff
@@ -425,7 +425,8 @@ def testAddOntology(self):
 
     def testRemoveOntology(self):
         ontologyName = "the_ontology_name"
-        cliInput = "remove-ontology {} {}".format(self.repoPath, ontologyName)
+        cliInput = "remove-ontology {} {}".format(
```
Comment (jeromekelleher) [inline tests/unit/test_cli.py]: Why the split? Looks like a fairly gratuitous change...

File: tests/end_to_end/test_repo_manager.py [hunk 3]
```diff
@@ -5,14 +5,11 @@
...
 
-import mock
 import os
-import shutil
 import tempfile
 import unittest
 
 import ga4gh.cli as cli
-import ga4gh.exceptions as exceptions
 import tests.paths as paths
```
Comment (jeromekelleher) [inline tests/end_to_end/test_repo_manager.py]: These changes are already covered in pr #1198 I think

File: docs/configuration.rst [hunk 4]
```diff
@@ -164,6 +199,33 @@ Removes a variant set from a given data repository and dataset.
 
     $ ga4gh_repo remove-variantset path/to/repo.db aDataset aVariantSet
 
+++++++++++++++
+add-featureset
+++++++++++++++
+
+Adds a feature set to the repo created using `scripts/generate_gff3_db.py`.
```
Comment (jeromekelleher) [inline docs/configuration.rst]: I'm not sure how helpful this actually is --- we don't distribute the ``generate_gff3_db.py`` script, so this is only going to cause confusion. We'll need to make this a first class CLI program, and document it separately. This needs to be done in a follow up issue, where we look at the requirements for importing featuresets properly and figure out the workflow for the administrator.

**Comment by jeromekelleher** [comment]
I'm afraid I don't see any of these changes as helping us much @david4096. The CLI documentation for repo manager needs a complete overhaul, and the changes here don't really improve it much. There's no point in adding in documenting a couple of options here and there; we need to comprehensively document the interface. It would be good if we could use something like [sphinx-argparse]
(http://sphinx-argparse.readthedocs.io/en/latest/) to do this, so that we only document the CLI once, in the source.

**Pull request closed without merging**